Firestore/Example/Tests/Integration/API/FIRServerTimestampTests.m

```diff
@@ -88,26 +101,58 @@ - (void)writeInitialData {
   XCTAssertEqualObjects(initialDataSnap.data, _initialData);
 }
 
-/** Waits for a snapshot containing _setData but with NSNull for the timestamps. */
-- (void)waitForLocalEvent {
-  FIRDocumentSnapshot *localSnap = [_accumulator awaitEventWithName:@"Local event."];
-  XCTAssertEqualObjects(localSnap.data, [self expectedDataWithTimestamp:[NSNull null]]);
+/** Waits for a snapshot with local writes. */
+- (FIRDocumentSnapshot *)waitForLocalEvent {
+  FIRDocumentSnapshot *snapshot = [_accumulator awaitEventWithName:@"Local event."];
+  XCTAssertTrue(snapshot.metadata.hasPendingWrites);
+  return snapshot;
+}
+
+/** Waits for a snapshot with that has no pending writes */
+- (FIRDocumentSnapshot *)waitForRemoteEvent {
+  FIRDocumentSnapshot *snapshot = [_accumulator awaitEventWithName:@"Remote event."];
+  XCTAssertFalse(snapshot.metadata.hasPendingWrites);
+  return snapshot;
+}
+
+/** Verifies a snapshot containing _setData but with NSNull for the timestamps. */
+- (void)verifySnapshotWithNullTimestamps:(FIRDocumentSnapshot *)snapshot {
```

Comment:
This name doesn't quite fit the style guide. The preposition "with" seems like you might be returning a snapshot but that's not what you're doing. `verifySnapshotHasNullTimestamps:` would address that part.

Also the label should end in "Snapshot" so you get that wonderful Objective-C feeling of repeating everything three times. Maybe `verifyNullTimestampsInSnapshot:`? That's a little clunky though.

Either of those would work for me (on this and the ones below).

Reply:
All methods are now renamed:
- verifyTimestampsAreNullInSnapshot
- verifyTimestampsAreEstimatedInSnapshot
- verifyTimestampsInSnapshot:FromPreviousSnapshot:

So happy that Apple fixed code completion with XCode 5....